lib/ansible/modules/crypto/openssl_csr.py

```diff
@@ -422,6 +442,7 @@ def main():
             keyUsage_critical=dict(aliases=['key_usage_critical'], default=False, type='bool'),
             extendedKeyUsage=dict(aliases=['extKeyUsage', 'extended_key_usage'], type='list'),
             extendedKeyUsage_critical=dict(aliases=['extKeyUsage_critical', 'extended_key_usage_critical'], default=False, type='bool'),
+            basicConstraints=dict(aliases=['basic_constraints'], type='list'),
```

Comment:
Would it be helpful if we provide user available types of basic constraints ?

Reply:
None of the other extension apis do this, it's also not entirely clear to me what constraints are supported.

Reply:
AFAIK, `critical,CA:TRUE, pathlen:0` can be included in example.

Reply:
I checked the [RFC](https://tools.ietf.org/html/rfc5280), and I think while `critical` is technically supported, it's not recommended for basicConstraints like it is for (`extended`)`keyUsage`. `critical` should also be set like `extendedKeyUsage_critical` and `keyUsage_critical`.

I also don't know if `pathlen` is a valid option. The RFC shows `pathLenConstraint`, but the RFC also doesn't necessarily match OpenSSL's behaviour and specifications: for example the RFC shows `cA`, but OpenSSL requires `CA`.